### PR TITLE
Book detail

### PR DESCRIPTION
**Pull request by Hayler27**
# Summary
### ·Like the book cover, I have converted the book detail into a web component.

# Trello Cards
###  https://trello.com/c/GZuGDuwg

# Screenshots
## Card 1:
Captura de Pantalla 2022-10-19 a la(s) 9 49 30 a m

## Card 2:
Captura de Pantalla 2022-10-19 a la(s) 9 49 38 a m

### CODE REVIEW

File: src/buscador.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  filterData(input) {
+    console.log(input, this.result);
+    this.result = [];
+    this.data.map((product) => {
+      const name = product.book_title.toLowerCase();
+
+      if (name.indexOf(input) !== -1) this.result = [...this.result, product];
+    });
```
Comment (dsandoval11) [inline src/buscador.js]: you can use `filter` function instead of map and set the value to `result` directly

File: src/card/card.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                  <div class="list-card">
+                    <div class="list-cover">
+                      <img class="img-cover" src="${this.cover.sprites?.img}" alt="${this.cover.author}">
+                      <h2 class="h2">${this.cover.author}</h2>
+                      <h1 class="h1">${this.cover.book_title}</h1>
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </div>
```
Comment (dsandoval11) [inline src/card/card.js]: fix the indentation

File: src/card/card.styles.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    flex-flow: column;
+    margin: 16px;
+    max-height: 300px;
+    max-width: 198px;
+    padding: 28px 28px 24px;
+    transition: transform 0.2s;
+
+    &:hover {
+      transform: scale(1.1);
+    }
```
Comment (dsandoval11) [inline src/card/card.styles.js]: You aren't using sass, I think this doesn't work 🤔

File: src/detail/detail.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+class DetailComp extends LitElement {
+
+  static get styles() {
+    return [styles];
+  }
+
+  static get properties() {
+    return {
+      cover: { book_title: Object }
```
Comment (dsandoval11) [inline src/detail/detail.js:13]: you can only use `type or `attribute`, it should be like this
```suggestion
      cover: { type: Object }
```

File: src/detail/detail.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    }
+  }
+
+  constructor() {
+    super();
+    this.cover = {
+      sprites: {},
+      author: "",
+      book_title: []
```
Comment (dsandoval11) [inline src/detail/detail.js:22]: Why the book_title is received in an array?

File: src/detail/detail.style.js [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+    height: 100%;
+    justify-content: center;
+    width: 100%;
+  
+    @include for-mobile {
+    flex-direction: column;
+    height: 667px;
+    margin: auto;
+    width: 376px;
+    }
```
Comment (dsandoval11) [inline src/detail/detail.style.js]: you can't use this in css

File: src/list/list.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      covers: { type: Object },
+      result: {typer: Array},
+    }
+  }
+
+  constructor() {
+    super();
+    this.covers = [];
+    this.result = [];
+    this.dataFromBuscador();
```
Comment (dsandoval11) [inline src/list/list.js]: don't mix English and Spanish 😬

File: src/list/list.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      detail: cover.author
+    });
+    this.dispatchEvent(event);
+  }
+  processData(data) {
+    if (data.lenght == 0) {
+      this.result= this.covers; 
+    } else {
+      this.result=data;
+    }
```
Comment (dsandoval11) [inline src/list/list.js]: use a ternary operator

File: src/buscador.js (hunk 1 above)
Comment (Hayler27) [reply src/buscador.js]: Done!

File: src/card/card.js (hunk 2 above)
Comment (Hayler27) [reply src/card/card.js]: Done!

File: src/card/card.styles.js (hunk 3 above)
Comment (Hayler27) [reply src/card/card.styles.js]: Done!

File: src/detail/detail.js (hunk 4 above)
Comment (Hayler27) [reply src/detail/detail.js:13]: Done!

File: src/detail/detail.js (hunk 5 above)
Comment (Hayler27) [reply src/detail/detail.js:22]: Done!

File: src/detail/detail.style.js (hunk 6 above)
Comment (Hayler27) [reply src/detail/detail.style.js]: Done!

File: src/list/list.js (hunk 7 above)
Comment (Hayler27) [reply src/list/list.js]: Done!

File: src/list/list.js (hunk 8 above)
Comment (Hayler27) [reply src/list/list.js]: Done!

File: src/card/card.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        <div class="list-cards">
+        <div class="list-card">
+        <div class="list-cover">
+          <img class="img-cover" src="${this.cover.sprites?.img}" alt="${this.cover.author}">
+          <h2 class="h2">${this.cover.author}</h2>
+          <h1 class="h1">${this.cover.book_title}</h1>
+        </div>
+        </div>
+        </div>
+        </div>
```
Comment (dsandoval11) [inline src/card/card.js]: Fix the indentation

File: src/searcher.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  filterData(input) {
+    console.log(input, this.result);
+    this.result = [];
+    this.data.filter((product) => {
+      const name = product.book_title.toLowerCase();
+
+      if (name.indexOf(input) !== -1) this.result = [...this.result, product];
+    });
```
Comment (dsandoval11) [inline src/searcher.js]: the `filter` function returns a new array with the filtered data, check this [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)

File: src/card/card.js (hunk 9 above)
Comment (Hayler27) [reply src/card/card.js]: Done!

**Review by dsandoval11 — APPROVED** [review]

**Pull request merged**